pennylane/bose/bosonic_mapping.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains functions to map bosonic operators to qubit operators."""
+
+from functools import singledispatch
+from typing import Union
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.pauli import PauliSentence, PauliWord
+
+from .bosonic import BoseSentence, BoseWord
+
+
+# pylint: disable=too-many-branches
+def _get_pauli_op(i, j, qub_id):
+    r"""Returns expression to convert qubit-local term ::math::``\ket{x_i}\bra{x_j}``
+    to qubit operators as given in :math:`Eq. (6-9)` in `arXiv.1909.12847 <https://arxiv.org/abs/1909.12847>_`.
+    """
+
+    c1, c2 = 0.5, -0.5 if i == 1 else 0.5
+
+    if i != j:
+        return PauliSentence({PauliWord({qub_id: "X"}): c1, PauliWord({qub_id: "Y"}): c2 * 1j})
+
+    return PauliSentence({PauliWord({}): c1, PauliWord({qub_id: "Z"}): c2})
+
+
+def binary_mapping(
+    bose_operator: Union[BoseWord, BoseSentence],
+    n_states: int = 2,
+    ps: bool = False,
+    wire_map: dict = None,
+    tol: float = None,
+):
+    r"""Convert a bosonic operator to a qubit operator using the standard-binary mapping.
+
+    The mapping procedure is described in equations: :math:`27-29` in `arXiv:1507.03271 <https://arxiv.org/pdf/1507.03271>`_.
+
+    Args:
+        bose_operator(BoseWord, BoseSentence): the bosonic operator
+        n_states(int): maximum number of allowed bosonic states
+        ps (bool): Whether to return the result as a ``PauliSentence`` instead of an
```

Comment:
For consistency we can:
```suggestion
        ps (bool): whether to return the result as a ``PauliSentence`` instead of an
```